### PR TITLE
delete lease and creds when you get a permission denied error

### PR DESCRIPTION
**Pull request by Joseph-Irving**
Currently if vault-creds gets into a state where its token has expired it will just constantly retry to renew while getting permission denied. This PR causes it to exit and delete the lease and creds file so when it starts back up it will generate a new one.

### CODE REVIEW

**Comment by Joseph-Irving** [comment]
Developers will still need to have their apps restart or reload the creds if they're getting permission denied on the db side

File: cmd/main.go [hunk 1]
```diff
@@ -153,10 +168,18 @@ func main() {
 				if err != nil {
 					log.Errorf("error renewing auth: %s", err)
 				}
+				if err == vault.ErrPermissionDenied {
```
Comment (combor) [inline cmd/main.go:171]: Is there a reason we don't put it here: https://github.com/uswitch/vault-creds/pull/12/files#diff-09487838846606ad96d627de95c1cb09R51 ?

Comment (Joseph-Irving) [reply cmd/main.go:171]: we need to cause the cleanup function to trigger which needs variables in the main.go

Comment (combor) [reply cmd/main.go:171]: Right.

**Review by combor — APPROVED** [review]

**Pull request merged**